### ISSUE TITLE
Codepen examples with React UMD not functional: `Uncaught TypeError: Cannot read property 'Component' of undefined`

### ISSUE THREAD

**Issue by perrinjerome** [rjsf-team/react-jsonschema-form#2503]
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

![image](https://user-images.githubusercontent.com/521510/128302805-0eabc040-33f2-4731-abcb-6e5ddfe01e47.png)

![image](https://user-images.githubusercontent.com/521510/128302840-e44a1aa5-c03a-4e12-a85a-7cbe2c5862f1.png)

I don't know exactly what's going on, but I noticed that if I include an extra script containing

```javascript
window.react = window.React;
```
between react dom and react-jsonschema-form, then this error disappear and the result is OK. 

### Steps to Reproduce

1. Open https://react-jsonschema-form.readthedocs.io/en/latest/#usage
2. Click on "Result" from Codepen widget

#### Expected behavior

A form should be generated from the example code

#### Actual behavior

A javascript error is logged in the console
```
ArrayField.js:199 Uncaught TypeError: Cannot read property 'Component' of undefined
    at Module.<anonymous> (ArrayField.js:199)
    at t (bootstrap:19)
    at bootstrap:83
    at react-jsonschema-form.js:1
    at universalModuleDefinition:9
    at universalModuleDefinition:1
(anonymous) @ ArrayField.js:199
t @ bootstrap:19
(anonymous) @ bootstrap:83
(anonymous) @ react-jsonschema-form.js:1
(anonymous) @ universalModuleDefinition:9
(anonymous) @ universalModuleDefinition:1
```

### Version

EXTERNAL JAVASCRIPT

https://cdnjs.cloudflare.com/ajax/libs/react/17.0.2/umd/react.production.min.js

https://cdnjs.cloudflare.com/ajax/libs/react-dom/17.0.2/umd/react-dom.production.min.js

https://unpkg.com/@rjsf/core/dist/react-jsonschema-form.js ( this is 3.0.0)

**Comment by perrinjerome** [comment]
This seems same as https://github.com/rjsf-team/react-jsonschema-form/issues/2498

Here as well, it works when changing to https://unpkg.com/@rjsf/core/dist/react-jsonschema-form.js to https://unpkg.com/@rjsf/core@2.1.0/dist/react-jsonschema-form.js

**Comment by epicfaace** [comment]
It looks like the latest version v3.0.1 (which should have fixed this
issue) wasn't released properly to npm, trying to fix in
https://github.com/rjsf-team/react-jsonschema-form/pull/2505.

--
Ashwin Ramaswami

**Comment by johnpaulkiser** [comment]
Is there any interest in using [codesandbox](https://codesandbox.io/docs/embedding) instead?

**Comment by epicfaace** [comment]
This should be fixed with 3.0.2 now! 🎉 

@johnpaulkiser codesandbox is also doable, though I'm not clear what we'd gain from switching codepen over to this. Are there any particular advantages you have in mind for it?

**Comment by perrinjerome** [comment]
Thank you @epicfaace
